lib/Doctrine/ORM/Query/SqlWalker.php

```diff
@@ -1388,7 +1388,12 @@ public function walkSelectExpression($selectExpression)
                 if (! $hidden) {
                     // Conceptually we could resolve field type here by traverse through AST to retrieve field type,
                     // but this is not a feasible solution; assume 'string'.
-                    $this->rsm->addScalarResult($columnAlias, $resultAlias, Type::getType('string'));
+                    if ($expr instanceof Query\AST\TypedExpression) {
```

Comment:
Maybe the comment above is no longer 100% accurate? Should probably be "assume 'string' when the expression is not typed"?